exec/exec.go

```diff
@@ -96,19 +101,44 @@ func New() Interface {
 	return &executor{}
 }
 
+// command is an internal method used to clear the error field from LookPath
+func (executor *executor) command(name string, arg ...string) *osexec.Cmd {
+	cmd2 := osexec.Command(name, arg...)
+	// Err was introduced in 1.19 only so we do this only if we find the field
+	field := reflect.ValueOf(cmd2).Elem().FieldByName("Err")
+	if field.IsValid() {
+		// osexec.Cmd caches the error from LookPath if any, so clear
+		// it to get the old behavior back.
+		err := (*error)(unsafe.Pointer(field.Addr().Pointer()))
+		*err = nil
+	}
```

Comment:
Don't really have an ask here, but putting it as a review to document a thought I had, maybe I missed something and someone else can add on to it:

Since this field was added in Go 1.19, instead of reflection, can we maybe make this decision by looking at the Go version being used using https://pkg.go.dev/runtime@go1.19rc2#Version?

I realised this might not be easy to do since `Version()` doesn't return a build tag always, and it can also return a commit hash along with date time or sometimes even values of `GOEXPERIMENT` env vars if set.

Reply:
@MadhavJivrajani y, i did not want to end up with permutations and combinations of parsing here. it may be even better to use go build tags, but then we'll need to keep updating the tag per go version.

this seemed to me as a safe alternative that we won't need to touch up repeatedly